src/openapi.ts

```diff
@@ -87,7 +106,7 @@ export const toOpenAPISpec = (
             in: 'query',
             name,
             description: schema.description,
-            schema: { type: 'string' },
```

Comment:
Wow. This was _definitely_ wrong 😂.

I'm not sure I'm following your other comment. Can we just pass `schema` (from line 85) directly here? Here's why I think that'd be better:

- It will preserve any non-type-level constraints on the schema like `min`, `max`, `pattern`, `minLength`, `maxLength`, etc.

I recognize that there is a "risk" that `schema` _might_ describe a type that can't be defined as a query parameter (e.g. `object`).

But, I think _that_ issue is outside the scope of this OpenAPI converter utility -- we can probably just assume that the type is acceptable.

Reply:
> Can we just pass schema (from line 85) directly here?

That was my first attempt. then realized that it will be lots of work because of the essential problem that `JSONSchema4` (value entry of `Request.properties`) is incompatible with `OpenAPIv3.SchemaObject`. I'm sure it could improve the `getSchemaObjectType()` to support both all primitive types and array/object types (just lots of work), I created this PR for PML-222 so I don't need to create a workaround.

Reply:
Would it be wrong to just cast the schema i.e. `schema: schema as OpenAPIV3.NonArraySchemaObjectType`?

Reply:
yes, it will pass the compiling/lint but fail the test case since `string` is the default type when not declared. it could be improved in next iteration to take care of this requirement

Reply:
> the essential problem that JSONSchema4 (value entry of Request.properties) is incompatible with OpenAPIv3.SchemaObject

This is "correct", for lack of a better term. It's actually an issue that's referenced throughout `openapi.ts`, and addressed via `@ts-expect-error`. The problem is not just at the type level -- OpenAPI supports a _slightly_ custom version of "json schema" spec that isn't perfectly compatible with standard JSONSchema.

The approach we've taken throughout the reset of the file is to explicitly ignore this issue via casting + `@ts-expect-error` calls, since the difference between OpenAPI schemas + standard JSONSchemas are extremely minor, and we rarely (if ever) encounter them.

I think I have a simple fix that I can follow-on with to improve the capability here.

Reply:
I wonder if a viable long term solution would be to drop use of the `JSONSchema4` package and only use types from `OpenAPIv3` in this repo. I don't have enough experience with this repo or those packages to know if that's a good idea though.